api/src/main/java/org/openmrs/module/kenyaemr/chore/AddCD11MappingCodeSetsForDiagnosis.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.chore;
+
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.ObjectNode;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.kenyacore.chore.AbstractChore;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Iterator;
+
+/**
+ * Update ICD 11 Mappings for Diagnosis
+ * Uses DiagnosisICD10ToICD11Mapper
+ */
+@Component("kenyaemr.chore.AddCD11MappingCodeSetsForDiagnosis")
+public class AddCD11MappingCodeSetsForDiagnosis extends AbstractChore {
+
+	/**
+	 * @see AbstractChore#perform(PrintWriter)
+	 */
+
+	@Override
+	public void perform(PrintWriter out) {
+
+		ObjectMapper mapper = new ObjectMapper();
+		ArrayNode conf = null;
+		try {
+			System.out.println("Retrieving ICD code mapper");
+			conf = (ArrayNode) mapper.readTree(icd11DiagnosisCodeJson());
+			if (conf != null) {
+				System.out.println("ICD code mapper found");
+				for (Iterator<JsonNode> it = conf.iterator(); it.hasNext(); ) {
+					ObjectNode node = (ObjectNode) it.next();
+
+					String conceptId = node.get("ConceptId").asText();
+					String icd11code = node.get("ICD11Code").asText();
+
+					String updateConceptSql = "Insert into concept_reference_map (\n" +
```

Comment:
To be honest, I thought we could use just the service API. SQL is a bit weird here

Reply:
I tried that - but the two tables concept_reference_map and concept_reference_term in use are not updated as we normally do not align them for new concepts as such many codes do not exists at the moment. This necessitated  a manual process.